### DISCUSSION TITLE
Make effects assigned to different CPU cores with help from cgroups?

### DISCUSSION THREAD

**Discussion by 38github** [rerdavies/ToobAmp#42] [Ideas]
Is this something you have considered? I am guessing it would mean a lot of coding but for users we could really use a lot more of the potential that our small SoC/SBC computers have. An option for a slot to be assigned to specific core maybe?

**Comment by rerdavies** [comment]
I have. 

In practice, it doesn't work well because, for the most part, audio flow is linear. The rare exception is when you have branches in your audio flow. However, audio plugins that do heavy-duty processing also tend to implement their own multithreading when processing, so the results can be less than desirable.  Also heavyweight audio plugins running in parallel tend to contend for cache space, so they run at far less than twice the speed. Machine-learning based plugins, in particular have this problem, as they are usually very close to perfectly cache-bound.

There are, admittedly, rare cases where this helps. e.g. running parallel instances of  circuit-simulation-based guitarix plugins.

It's a possible future feature, but not currently a high-priority feature.

For what it's worth, the convolution reverb and convolution IR plugins use multithreading already. I don't think the ML plugins would benefit.
